mutation_query.hh

```diff
@@ -121,6 +121,8 @@ class reconcilable_result_builder {
     const schema& _schema;
     const query::partition_slice& _slice;
     bool _reversed;
+    schema_ptr _query_schema_storage; // Engaged only if _reversed.
+    const schema& _query_schema; // If _reversed, equal to *_query_schema_storage. Otherwise, equal to _schema.
```

Comment:
This is unneeded complication, considering the small difference between a reference and a shared ptr instance. Just have a `schema_ptr _query_schema` always.

Reply:
I wanted to do that, but how do I initialize the `schema_ptr _query_schema` if the query isn't reversed? The constructor doesn't take a `schema_ptr`, only a `const schema&`.

So I see 4 options:
1. Change the constructor (and recursively all its callers) to pass `schema_ptr` instead of `const schema&).
2. Convert the reference to a `shared_ptr` via `shared_from_this()`. But is this legal? How do I know it's always legal?
3. Copy the schema, which IIUC isn't that cheap. (We do it in the reverse case, but I guess we would like to avoid it in the non-reverse case).
4. Do what I did.

Which should I choose?

Reply:
> 2. Convert the reference to a shared_ptr via shared_from_this(). But is this legal? How do I know it's always legal?

This is the way to go. This is always legal. All schemas are shared ptr.

In general, if a class offers, `shared_from_this()`, it **must** always be legal to use it, otherwise it is useless.

Reply:
Alright.